@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -246,17 +286,28 @@ export class SimpleTileGeometryLoader implements TileGeometryLoader {
         enabledKinds: GeometryKindSet | undefined,
         disabledKinds: GeometryKindSet | undefined
     ) {
+        if (this.isFinished) {
+            return;
+        }
         // If the tile is not ready for display, or if it has become invisible while being loaded,
         // for example by moving the camera, the tile is not finished and its geometry is not
         // created. This is an optimization for fast camera movements and zooms.
         const tile = this.tile;
         const decodedTile = this.m_decodedTile;
-        this.m_decodedTile = undefined;
+        // TODO: Entire block could be moved to update method, enclose visibility check and
+        // disposal in one function used in update and timeout.
         if (decodedTile === undefined || tile.disposed || !tile.isVisible) {
+            // TODO: Should we dispose tile here?
+            //if (!tile.isVisible) {
+            //    tile.mapView.visibleTileSet.disposeTile(tile);
+            //}
             this.finish();
             return;
         }
+
         this.m_timeout = setTimeout(() => {
+            this.m_timeout = undefined;
```

Comment:
In my opinion we should not misuse the browser task queue as tile load queue.
This would all be much cleaner and more flexible if we would have a dedicated queue were we push new tiles to and only have the setTimeout fetching one tile from the queue.

In this way we would also be able to take the tile with the highest priority from the queue in the future.

Reply:
I fully agree, just keep in mind that this PR have been introduced to fix multiple buildings extrusion animation that we spot on Firefox..., just look where it goes. Seems that in buildings fix I will need to provide entire new architecture, both with dropping some old features (Phased Loading?). Do we really want to go this way?
Maybe separate tickets should be created for all that steps..

Reply:
Separate ticket created for it.